src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.String.cs

```diff
@@ -73,22 +71,16 @@ public static object Deserialize(string json, Type returnType, JsonSerializerOpt
             object result;
             byte[] tempArray = null;
 
-            int maxBytes;
+            // For performance, avoid obtaining the actual byte count unless the memory usage may be
+            // higher than the threshold which is options.DefaultBufferSize.
+            long maxBytes = (long)json.Length * JsonConstants.MaxExpansionFactorWhileTranscoding;
```

Comment:
It doesn't look like we need this anymore, especially the long.

How about this?
```C#
Span<byte> utf8 = json.Length <= (ArrayPoolMaxSizeBeforeUsingNormalAlloc / MaxExpansionFactorWhileTranscoding) ?
    tempArray = ArrayPool<byte>.Shared.Rent(json.Length * MaxExpansionFactorWhileTranscoding) :
    new byte[JsonReaderHelper.GetUtf8ByteCount(json.AsSpan())];
```

[sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgJgIwCwAUFgAwAEWeAdAEICuAZo4lANzFmU0AqcAHjA5FOAZko5yAYXIBvYuUXkADggCWANwjxyAYwD2AO1jk1hmOQCyEfgFF+yiMbVGAYhF0x9CAOoALNQAbOB4EJygDABMzAHNyAF5yUWElPSMTQKM4gEEEMIBPAAV9fUDrfgBlNQAvOFo4Rm84AFUoWIA5b0hA7MDM3QTyPFIcABZyACohkdGUpQUlLHEzC0sACgqYdUM4gCsoIwBKOQXUxWB8+ABtAF1yeDBlXILBw3o+ubPyTJ3ySH5aJc4FBBmsfjFDvsjNQADJwHYwPyTKw2eyOZxuDxeXwBYKhcJRWKfM4VdEAHgu8AAfOR6DBGAAOQb/QHwEFkxLPCBFEplGxVWr1RoIFptHadBDdXr9cgAflOXyUDyeeW5gy5PNKFKBVOoFT8EBFkWoACV4TA1msVocWUCoMcQArFeRDHAAO7kSlwK62QyEnbUZo8VwM6gAcTgMFZcCk+no5jWUMM1GyUFJTjWh0ON2JqSdiwA7LT6QyrqQcwqAL4KhVLUzmKw4DZbWLkJPHeREZ1e273OCPDWvd6BXNKdOGbXU4uMwZJ2HwmKI8gc8hrDXFUrlAV1BpNVodLoQHp9fQDAD0KLsDnCLkM7k83n8QRCYWM/ohcvzX2Vg85qs1gSTnAur6oacDGmaCZznCCJIlM5RojemIPjiz74m++jRDsDpfmcroej2vrvoGwahhGUZArG8YWnOqbjpm2ajoouFYEWdKMmWFZdko1ZELxQA==)

Reply:
Sure. Originally needed to worry about overflow; also tried to prevent both * 3 and / 3 however the IL is about the same so changing it to suggested.